### PR TITLE
[Communication] - #2hqq5my - [ValueFirst] Create Library for ValueFirst for OTP

### PR DESCRIPTION
**Pull request by fairyhunter13**
# Changelog
- Initial commit :tada:
- Add functional options :sparkles:
- Add enable token, generate token, and disable token actions :sparkles:
- Add request and response struct for SendSMS :sparkles:
- Adding client to send request to the ValueFirst :sparkles:
- Add Makefile for easier management of package :sparkles::hammer:
- Define unknown error for response that are not documented :sparkles:
- Add delete token action :sparkles:
- Add static check for interface :rotating_light:
- Working on Send SMS action :sparkles:
- Adding const of URLActionDeleteToken :sparkles:
- Add err for nil arguments in the SendSMS method :sparkles:
- Upgrade reflecthelper from v4 to v5 :package::arrow_up:
- Add more method to the Client interface :sparkles:
- Add custom IPs feature :sparkles:
- Update go modules :package:
- Add token manager :sparkles:
- Add TODO for todo list :memo:
- Add redis as storage hub for token :sparkles:
- Add logic for Token manipulation :sparkles:
- Refactor interface Client to move some methods to manager.TokenClient :recycle:
- Move some struct to the manager package :recycle:
- Refactor some naming :recycle:
- Completing send SMS request logic :sparkles:
- Add local storage hub :sparkles:
- Add redis storage hub :sparkles:
- Fix token manager to enable token before saving to the storage :bug:
- Add token manager and token storage in the Option :sparkles:
- Working on GetError logic :construction:
- Refactor token manager to be initialized using functional options :recycle:
- Add .env example for testing :wrench:
- Refactor and move some structs to the other files :recycle:
- Change ResponseMessageAckGUIDError and ResponseMessageAckGUIDErrors to Error interface :sparkles:
- Add integration tests that can only run on local :white_check_mark:
- Add static checks :rotating_light:
- Change some error messages :goal_net:
- Refactor some private functions :recycle:
- Add an integration tests because of complex API :white_check_mark:
- Remove omitempty in the request to conform DTD standard :bug:
- Replace native type with the newtype :sparkles:
- Add LICENSE :page_facing_up:
- Update README.md :memo:
- Deleting local test files :fire:

ClickUp: https://app.clickup.com/t/2hqq5my

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,5 @@
+VALUEFIRST_USERNAME={{your valuefirst username}}
+VALUEFIRST_PASSWORD={{your valuefirst password}}
+VALUEFIRST_IP={{your valuefirst ip}}
```
Comment (asqolani-flip) [inline .env.example:3]: ip ini dari kita bukan mas?

File: constants.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package valuefirst
+
+// List of URLs used in this package.
+const (
+	URLToken               = "/api/messages/token"
+	URLActionGenerateToken = URLToken + "?action=generate"
+	URLActionEnableToken   = URLToken + "?action=enable"
+	URLActionDisableToken  = URLToken + "?action=disable"
+	URLActionDeleteToken   = URLToken + "?action=delete"
+	URLActionSendSMS       = "/servlet/psms.JsonEservice"
```
Comment (asqolani-flip) [inline constants.go:10]: knp const-nya tidak pakai huruf besar semua mas? apakah ada concern?

File: .env.example (hunk 1 above)
Comment (fairyhunter13) [reply .env.example:3]: @asqolani-flip ya mas

File: constants.go (hunk 2 above)
Comment (fairyhunter13) [reply constants.go:10]: @asqolani-flip 

> Naming Conventions for Golang Constants
> - Name of constants must follow the same rules as variable names, which means a valid constant name must starts with a letter or underscore, followed by any number of letters, numbers or underscores.
> - By convention, constant names are usually written in uppercase letters. This is for their easy identification and differentiation from variables in the source code.

References:
- https://www.golangprograms.com/go-language/constants.html
- https://go.dev/blog/constants

**Review by asqolani-flip — APPROVED** [review]

File: constants.go (hunk 2 above)
Comment (asqolani-flip) [reply constants.go:10]: iya mas biasanya pakai uppercase underscore.

**Pull request merged**